asdf.sh

```diff
@@ -33,6 +33,7 @@ PATH="${ASDF_USER_SHIMS}:$PATH"
 . "${ASDF_DIR}/lib/asdf.sh"
 
 if [ -n "${ZSH_VERSION:-}" ]; then
-  autoload -U bashcompinit
-  bashcompinit
+  fpath=(${ASDF_DIR}/completions $fpath)
+  autoload -Uz compinit
+  compinit
```

Comment:
Use `compinit` as per latest instructions in README.

in #69 we changed to `bashcompinit` for zsh users in this script, but then in #270 instructed users to use `compinit` in the README. We will go with the zsh communities latest suggestion unless someone raises an issue with this change?

Reply:
This was a bad decision. See #674 & fix in #675